maven-release-manager/src/main/java/org/apache/maven/shared/release/DefaultReleaseManager.java

```diff
@@ -107,23 +107,7 @@ private void prepare( ReleasePrepareRequest prepareRequest, ReleaseResult result
         final ReleaseDescriptorBuilder builder = prepareRequest.getReleaseDescriptorBuilder();
 
         // Create a config containing values from the session properties (ie command line properties with cli).
-        ReleaseUtils.copyPropertiesToReleaseDescriptor( prepareRequest.getUserProperties(),
-                                        new ReleaseDescriptorBuilder()
-                                        {
-                                            public ReleaseDescriptorBuilder addDevelopmentVersion( String key,
-                                                                                                   String value )
-                                            {
-                                                builder.addDevelopmentVersion( key, value );
-                                                return this;
-                                            }
-
-                                            public ReleaseDescriptorBuilder addReleaseVersion( String key,
-                                                                                               String value )
-                                            {
-                                                builder.addReleaseVersion( key, value );
-                                                return this;
-                                            };
-                                        } );
+        ReleaseUtils.copyPropertiesToReleaseDescriptor( prepareRequest.getUserProperties(), builder );
```

Comment:
original code should be extended with

                        @Override
                        public ReleaseDescriptorBuilder addDependencyDevelopmentVersion( String dependencyKey,
                                                                                         String version )
                        {
                            builder.addDependencyDevelopmentVersion( dependencyKey,
                                                                     version );
                            return this;
                        };
    
                        @Override
                        public ReleaseDescriptorBuilder addDependencyReleaseVersion( String dependencyKey,
                                                                                     String version )
                        {
                            builder.addDependencyReleaseVersion( dependencyKey,
                                                                 version );
                            return this;
    
                        };

This way we have full control over the properties we want to allow to be overwritten via commandline.

Reply:
Hi Robert,
I agree - I have added the three new methods `addDependencyOriginalVersion`, `addDependencyReleaseVersion`, `addDependencyDevelopmentVersion`.